### DISCUSSION TITLE
Pending request for a long time

### DISCUSSION THREAD

**Discussion by pbussi** [deepgram/community#228] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

I am using the API for a platform and it works fine. But, in some cases, with long video files (~2hours), the request is pending forever. No errors were produced. When the api is requested, the req_id is generated with no problem.
Did this happened to anyonone? Any help will be appreciated.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?callback=https://stagingnode.theirstory.io%2Ftranscripts/648b205b7c14b1967895a67e&language=en&diarize=true&summarize=true&detect_topics=true&detect_entities=true&model=whisper-large&numerals=false

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

407e0089-6df5-4fe4-823b-185b184ebc3c

**Comment by jkroll-deepgram** [comment]
Hi @pbussi, I'm curious how long you are waiting for a response? Whisper has higher latency, and the `whisper-large` model could take several minutes or more for a 2-hour batch transcription. You could give our much faster Nova model a try (`model=nova`), which would take around ~15-30 seconds for a 2-hour batch.

**Reply by pbussi** [reply]
Hi @jkroll-deepgram , thank for your response. Unfortunately, it didnt work. I tried the model=nova, but same behavour (req id: ace28993-0d0d-4b9f-ad97-fe75e58282b1). BTW, the original request is still pending.
This is the curl-like request i used:

`curl -X POST -H "Authorization: Token xxx" -H 'content-type: application/json' -d '{"url":"https://theirstory.s3.us-west-1.amazonaws.com/uploads/xxx"}' "https://api.deepgram.com/v1/listen?callback=https://stagingnode.theirstory.io%2Ftranscripts/648b205b7c14b1967895a67e&language=en&diarize=true&summarize=true&detect_topics=true&detect_entities=true&model=nova&numerals=false"`
Sorry, I cant make the file public

**Reply by jkroll-deepgram** [reply]
Hi @pbussi, I see on Deepgram's side that the Nova request returned the callback result after ~40 seconds on June 26, and the Whisper request returned the callback result after ~2 minutes 15 seconds on June 15. Does that match with when you submitted the requests? Are you seeing any details from callbacks received on your side? You may find helpful details on our [Callback Not Recieved](https://developers.deepgram.com/docs/payload-too-large) doc as well. Possibly the callback payload is too large for the default configuration of your server.

**Reply by pbussi** [reply]
Hi, yes, I checked the payload body limit and it is set to 50MB. In fact, today I checked using my local environment callback using ngrok and no response from deepgram. I checked also using a base model, just in case.
req_id: 6e0455b5-bd78-4d7f-8221-d8224d727f51
The weird thing here, i had callback invocation errors before. In these cases, deepgram show code errors (usually 500) and # of retries (up to 10). But the cases I am looking for, the attempts value is set to 0 (deepgram didnt attempt to invoke the callback)

**Reply by pbussi** [reply]
Right now, i am seeing operations in pending status. Now i prepared the callback to process the errors, and for the pending status operations, i got a call with this body:
`err_code: "Gateway Timeout"
err_msg: "Request timeout. Please try again later."
request_id: "0dd04d08-95c9-4f80-920b-4967d9aca74f"`

What does this mean? Deepgram will call the callback again? And why the log show it in pending status?

**Reply by pbussi** [reply]
@jkroll-deepgram , is it gateway timeout am error produced because dg takes a lot to download the file? We are using s3 objects.

**Reply by jkroll-deepgram** [reply, marked as answer]
Hi @pbussi, as a next step, I recommend converting a video file to audio before sending it to Deepgram, to avoid timeouts or prolonged processing times. You can do this on the command line with ffmpeg, like `ffmpeg -i <filename>.mp4 <filename>.mp3`. If you do an initial test and that works, you can also look into [Amazon Elastic Transcoder](https://aws.amazon.com/elastictranscoder/) to bulk-convert video into audio.

I see you're using several of our understanding features (diarize, summarize, detect topics, detect entities) - these involve a longer processing time as well. If the conversion to audio doesn't help, you could also look into dividing very long files into smaller, ~1 hour chunks for processing.

**Reply by pbussi** [reply]
Thank you for your reply! I will try to reduce time by disabling summarize, topics and entities detection. Will perform conversion if the timeout cannot be solved.
